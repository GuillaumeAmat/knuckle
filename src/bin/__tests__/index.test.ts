import packageJson from '../../../package.json';
import { binScriptPath } from '../../helpers/tests/binScriptPath';
import * as cmd from '../../helpers/tests/cmd';
import { setupPristineTestFolder } from '../../helpers/tests/setupPristineTestFolder';

describe('Knuckle', () => {
  setupPristineTestFolder();

  it("should print Knuckle's version", async done => {
    let response = await cmd.run('node', [binScriptPath, '-v']) || undefined;
    expect(response?.trim()).toBe(packageJson.version);

    response = await cmd.run('node', [binScriptPath, '--version']) || undefined;
    expect(response?.trim()).toBe(packageJson.version);

    done();
  });
});

describe('Usage', () => {
  setupPristineTestFolder();

  it('should print the usage', async done => {
    let response = await cmd.run('node', [binScriptPath]);
    expect(response).toMatchSnapshot();

    response = await cmd.run('node', [binScriptPath, '-h']);
    expect(response).toMatchSnapshot();

    response = await cmd.run('node', [binScriptPath, '--help']);
    expect(response).toMatchSnapshot();

    done();
  });
});

describe('Error', () => {
  setupPristineTestFolder();

  it('should print an `unknown option` error', async done => {
    try {
      await cmd.run('node', [binScriptPath, '--bad-option']);
    } catch (err) {
      expect(err.trim()).toMatchSnapshot();
    }

    expect.hasAssertions();

    done();
  });
});
