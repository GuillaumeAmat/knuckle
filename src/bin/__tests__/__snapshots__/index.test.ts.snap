// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error should print an \`unknown option\` error 1`] = `""`;
